pkg/converter/target/utils.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package target
+
+import (
+	"antrea.io/nephe/pkg/controllers/config"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetExternalEntityKeyFromSource(source ExternalEntitySource) client.ObjectKey {
+	access, _ := meta.Accessor(source)
+	return client.ObjectKey{Namespace: access.GetNamespace(),
+		Name: getTargetEntityName(source.EmbedType(), false)}
+}
+
+func GetExternalNodeKeyFromSource(source ExternalNodeSource) client.ObjectKey {
+	access, _ := meta.Accessor(source)
+	return client.ObjectKey{Namespace: access.GetNamespace(),
+		Name: getTargetEntityName(source.EmbedType(), true)}
+}
+
+// GetExternalEntityLabelKind returns value of ExternalEntity kind label.
+func GetExternalEntityLabelKind(obj runtime.Object) string {
+	return strings.ToLower(reflect.TypeOf(obj).Elem().Name())
+}
+
+// getTargetEntityName returns the desired name of the target resource.
+func getTargetEntityName(obj runtime.Object, externalNode bool) string {
+	access, _ := meta.Accessor(obj)
+	if externalNode {
+		// ExternalNode CR name will be vm-<VirtualMachine CR name>.
+		return "vm-" + strings.ToLower(access.GetName())
+	}
+	// ExternalEntity CR name will be virtualmachine-<VirtualMachine CR name>.
```

Comment:
I would like to see both external entity and external node have same prefix as vm, rather than virtualmachine-

Reply:
Sure

Reply:
I have updated the code. Integration tests may also need modification to generate ExternalEntity with name as vm instead of virtualmachine.